packages/cli/src/deploy/getPublicLibraries.ts

```diff
@@ -0,0 +1,79 @@
+import { readFile } from "fs/promises";
+import { getCreate2Address, keccak256, stringToHex } from "viem";
+import toposort from "toposort";
+import glob from "glob";
+import { LinkReferences, getContractData } from "../utils/utils/getContractData";
+import { deployer } from "./ensureDeployer";
+import { PublicLibrary, salt } from "./common";
+import path from "path";
+
+export async function getPublicLibraries(forgeOutDir: string) {
+  const libraryDeps: {
+    libraryFilename: string;
+    libraryName: string;
+    contractFullyQualifiedName: string;
+    libraryFullyQualifiedName: string;
+  }[] = [];
+  const files = glob.sync(`${forgeOutDir}/**/*.json`, { ignore: "**/*.abi.json" });
+
+  for (const contractOutPath of files) {
+    const json = JSON.parse((await readFile(contractOutPath, "utf8")).trim());
+    const linkReferences = json.bytecode.linkReferences as LinkReferences;
+
+    const contractFullPath = Object.keys(json.metadata.settings.compilationTarget)[0];
```

Comment:
Previously this fetched `contractFullPath` [from `json.ast.absolutePath;`](https://github.com/latticexyz/mud/commit/5f5a0c156515c66f59029ce6fc9c46d7e410d66f), but the ABI's don't have that `ast` property anymore. Maybe Foundry removed it?

I don't love this change but it seems to work 🤔

Reply:
this is actually the first version I had
later I switched it to ast cause an array deep in input metadata felt less reliable, but either way works fine afaik
we could probably add `ast` back to output, but that would require overriding forge defaults, which seems worse

Reply:
ah interesting, I didn't know there was a Forge option for this. I will try it and see what feels best 🤔

Reply:
since compilationTarget is an input to solc, this might be a good question for foundry devs or someone familiar with its source
I assume it only ever has a single key, and ur approach is fine

Reply:
I am seeing this when running tests:
```
│ /Users/kevin/Projects/latticexyz/mud/packages/cli/src/deploy/getPublicLibraries.ts:22
│     const contractFullPath = Object.keys(json.metadata.settings.compilationTarget)[0];
│                                                        ^
│ TypeError: Cannot read properties of undefined (reading 'settings')
```